feature-flags/src/redis.rs

```diff
@@ -40,38 +58,44 @@ impl Client for RedisClient {
 
         Ok(fut?)
     }
-}
 
-// TODO: Find if there's a better way around this.
-#[derive(Clone)]
-pub struct MockRedisClient {
-    zrangebyscore_ret: Vec<String>,
-}
+    // TODO: Ask Xavier if there's a better way to handle this.
```

Comment:
looking for suggestions here

Reply:
Maybe you could downcast from anyhow? something like
```rust
impl From<anyhow::Error> for FlagError {
    fn from(error: anyhow::Error) -> Self {
        match error.downcast::<CustomRedisError>() {
            Ok(CustomRedisError::NotFound) => FlagError::NotFound,
            Err(_) => FlagError::InternalServerError,           
        }        
    }
}
```
from what I could understand you would like to error when redis returns empty for a key and the other errors you bubble up to the api handler without the need to customise each one, and then let the "FlagError into response impl" assign the http status based on what you logic implies. 

what I did in the past was something like this https://github.com/thiagovarela/rust-minijinja-htmx/blob/main/server/src/error.rs#L16-L47, without thiserror, and manually implementing whatever specialised error I'd get.

Reply:
ooh nice, thank you @thiagovarela , this is what I was missing! - implementing the From trait here, and downcasting (what I meant by reverse-coerce lol).

I think I want to handle these errors a little differently, so I'll keep the CustomRedisError, but good to know this is how I'd do it implicitly.

Reply:
- Thinking of the `redis` module as an isolated  thing, it makes sense for it to return it's own error type like you did. You can then implement `impl From<CustomRedisError> for FlagError` for the `?` operator to work transparently. This would indeed be the cleanest way. Make sure that the `match` in your `From` impl does not use the `_` fallback, so that you remember to update it when new enum variants are added
- A shortcut would be for `RedisClient` to directly return `FlagError`s, like [we do for capture sinks returning CatpureErrors](https://github.com/PostHog/hog-rs/blob/4db3670622086546ffc9b37fcf1bca25490cb0c9/capture/src/sinks/mod.rs#L10). Not the best module isolation, but can be refactored later on

As we probably will want to extract the `RedisClient` in a common crate, I'd go with a separate error type though.

Reply:
Nice, thanks!

Yeah, agree, I didn't want to just return FlagErrors from redis. I feel a lot more comfortable doing this now that the type system will protect me from unknown thrown errors.